src/app/3d/qgs3ddebugwidget.cpp

```diff
@@ -112,19 +121,75 @@ void Qgs3DDebugWidget::setMapSettings( Qgs3DMapSettings *mapSettings )
   {
     mMap->setDebugDepthMapSettings( mDebugDepthMapGroupBox->isChecked(), static_cast<Qt::Corner>( mDebugDepthMapCornerComboBox->currentIndex() ), value );
   } );
+
+  // connect signals emitted by castCameraInputValue
+  connect( this, &Qgs3DDebugWidget::nearPlaneChanged, m3DMapCanvas->cameraController()->camera(), &Qt3DRender::QCamera::setNearPlane );
+  connect( this, &Qgs3DDebugWidget::farPlaneChanged, m3DMapCanvas->cameraController()->camera(), &Qt3DRender::QCamera::setFarPlane );
+  connect( this, &Qgs3DDebugWidget::positionChanged, m3DMapCanvas->cameraController()->camera(), &Qt3DRender::QCamera::setPosition );
 }
 
-/**
-   * Update the state of navigation widget from camera's state
-   */
 void Qgs3DDebugWidget::updateFromCamera() const
 {
-  mNearPlane->setValue( m3DMapCanvas->cameraController()->camera()->nearPlane() );
-  mFarPlane->setValue( m3DMapCanvas->cameraController()->camera()->farPlane() );
-  mCameraX->setValue( m3DMapCanvas->cameraController()->camera()->position().x() );
-  mCameraY->setValue( m3DMapCanvas->cameraController()->camera()->position().y() );
-  mCameraZ->setValue( m3DMapCanvas->cameraController()->camera()->position().z() );
-  mLookingX->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().x() );
-  mLookingY->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().y() );
-  mLookingZ->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().z() );
+  whileBlocking( mNearPlane )->setValue( m3DMapCanvas->cameraController()->camera()->nearPlane() );
+  whileBlocking( mFarPlane )->setValue( m3DMapCanvas->cameraController()->camera()->farPlane() );
+  whileBlocking( mCameraX )->setValue( m3DMapCanvas->cameraController()->camera()->position().x() );
+  whileBlocking( mCameraY )->setValue( m3DMapCanvas->cameraController()->camera()->position().y() );
+  whileBlocking( mCameraZ )->setValue( m3DMapCanvas->cameraController()->camera()->position().z() );
+  whileBlocking( mLookingX )->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().x() );
+  whileBlocking( mLookingY )->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().y() );
+  whileBlocking( mLookingZ )->setValue( m3DMapCanvas->cameraController()->lookingAtPoint().z() );
+}
+
+void Qgs3DDebugWidget::castCameraInputValue( const double value )
+{
+  if ( sender() == mCameraX || sender() == mCameraY || sender() == mCameraZ )
+  {
+    auto *newPosition = new QVector3D( m3DMapCanvas->cameraController()->camera()->position().x(), m3DMapCanvas->cameraController()->camera()->position().y(), m3DMapCanvas->cameraController()->camera()->position().z() );
+    if ( sender() == mCameraX )
+    {
+      newPosition->setX( static_cast<float>( value ) );
+      emit positionChanged( *newPosition );
```

Comment:
there's no need to define the `positionChanged`, `nearPlaneChanged` and `farPlaneChanged` signals in this class - you can simply call `m3DMapCanvas->cameraController()->camera()->setPosition( newPosition )` instead...